test/integration/smoke/test_direct_download.py

```diff
@@ -184,20 +184,28 @@ def setUpClass(cls):
                 cls.services["service_offerings"]["tiny"]
             )
             cls._cleanup.append(cls.service_offering)
-            cls.network_offering = NetworkOffering.create(
-                cls.apiclient,
-                cls.services["l2-network_offering"],
-            )
-            cls.network_offering.update(cls.apiclient, state='Enabled')
-            cls.services["network"]["networkoffering"] = cls.network_offering.id
-            cls.l2_network = Network.create(
-                cls.apiclient,
-                cls.services["l2-network"],
-                zoneid=cls.zone.id,
-                networkofferingid=cls.network_offering.id
-            )
-            cls._cleanup.append(cls.l2_network)
-            cls._cleanup.append(cls.network_offering)
+
+            if cls.zone.networktype == 'Basic' :
+                networks = Network.list(cls.apiclient)
+                if len(networks) == 0 :
+                    self.skipTest("Skipping test since no network found in basic zone")
+                else :
+                    cls.network = networks[0]
+            else :
+                cls.network_offering = NetworkOffering.create(
+                    cls.apiclient,
+                    cls.services["l2-network_offering"],
+                )
+                cls.network_offering.update(cls.apiclient, state='Enabled')
+                cls.services["network"]["networkoffering"] = cls.network_offering.id
+                cls.network = Network.create(
+                    cls.apiclient,
+                    cls.services["l2-network"],
+                    zoneid=cls.zone.id,
+                    networkofferingid=cls.network_offering.id
+                )
+                cls._cleanup.append(cls.network)
+                cls._cleanup.append(cls.network_offering)
```

Comment:
these should be reversed and nearer to the respective creations
```suggestion
                cls.network_offering = NetworkOffering.create(
                    cls.apiclient,
                    cls.services["l2-network_offering"],
                )
                cls._cleanup.append(cls.network_offering)
                cls.network_offering.update(cls.apiclient, state='Enabled')
                cls.services["network"]["networkoffering"] = cls.network_offering.id
                cls.network = Network.create(
                    cls.apiclient,
                    cls.services["l2-network"],
                    zoneid=cls.zone.id,
                    networkofferingid=cls.network_offering.id
                )
                cls._cleanup.append(cls.network)
```

and also the tearDownClass should read:

```
    @classmethod
    def tearDownClass(cls):
        super(TestDirectDownloadTemplates, cls).tearDownClass()
```

(could be a separate PR but I tend to touch no test files without these kind of changes)

Reply:
and btw 
```
    def tearDown(self):
        super(TestDirectDownloadTemplates, swlf).tearDown()
```